django/tasks/backends.py

```diff
@@ -0,0 +1,119 @@
+from . import constants
+from .base import get_backend
+from .exceptions import ResultUnavailable
+
+
+class TaskResult(object):
+    """
+    Wrapper class representing a status of a task. Provides methods to retrieve
+    the task's status and result value/error.
+    """
+    def __init__(self, backend, task_id):
+        if not isinstance(backend, BaseBackend):
+            backend = get_backend(backend)
+        self._backend = backend
+        self.task_id = task_id
+
+    def __repr__(self):
+        return '<TaskResult %s - %r>' % (self.alias, self.task_id)
+
+    @property
+    def alias(self):
+        return self._backend.alias
+
+    def get_status(self, **kwargs):
+        """
+        Retrieve the status of the task from the backend.
+        """
+        return self._backend.status(self.task_id, **kwargs)
+
+    def get_result(self, **kwargs):
+        """
+        Retrieve the result of the task from the backend if the task has
+        successfully finished or the causing error if the task has failed.
+
+        raises a ResultUnavailable exception when there is no result available:
+            ResultUnknown if status is UNKNOWN
+            ResultPending if status is PENDING
+        """
+        return self._backend.get_result(self.task_id, **kwargs)
+
+
+class BaseBackend(object):
+    def __init__(self, alias, **kwargs):
+        self.alias = alias
+
+    def get_status(self, task_id):
+        """
+        The current status of the task.
+
+        PENDING: the task is currently in-queue, and waiting to be ran.
+        FAILED: the task attempted to run, but failed for some reason
+        SUCCESS: the task successfully ran
+        UNKNOWN: unable to determine the status of the task
+        """
+        raise NotImplementedError
```

Comment:
Add a exception message: "Subclasses of BaseBackend need to implement this method" or something like that.